pycaret/loggers/dagshub_logger.py

```diff
@@ -0,0 +1,102 @@
+import os
+from pathlib import Path
+
+from pycaret.loggers.mlflow_logger import MlflowLogger
+
+try:
+    import dagshub
+    import mlflow
+except ImportError:
+    dagshub = None
+
+
+class DagshubLogger(MlflowLogger):
+    def __init__(self, remote=None) -> None:
+        if dagshub is None:
+            raise ImportError(
+                "DagshubLogger requires dagshub. Install using `pip install dagshub`"
+            )
+        super().__init__()
+
+        # check token exist or not:
+        from dagshub.auth.tokens import TokenStorage, _get_token_storage
+
+        token_dict = _get_token_storage()._load_cache_file()
+        is_token_set = not TokenStorage._is_expired(token_dict)  # Check OAuth
+        if not is_token_set:
+            dagshub.auth.get_token()
+
```

Comment:
These are privet functions, instead we should use the dagshub.common.config to check if the user is logged in.